shell/platform/android/image_external_texture_gl.cc

```diff
@@ -36,26 +36,38 @@ ImageExternalTextureGL::ImageExternalTextureGL(
 
 void ImageExternalTextureGL::Attach(PaintContext& context) {
   if (state_ == AttachmentState::kUninitialized) {
-    if (!android_image_.is_null()) {
+    if (!android_image_.is_null() && !last_bounds_.isEmpty()) {
+      // After detach the cache of textures will have been cleared. If
+      // there is an android image we must populate it now so that the
+      // first frame isn't blank.
       JavaLocalRef hardware_buffer = HardwareBufferFor(android_image_);
-      AHardwareBuffer* hardware_buffer_ahw =
+      AHardwareBuffer* latest_hardware_buffer =
```

Comment:
A lot of this code is duplicated from ProcessFrame, can it be factored in such a way that we can reuse the code? The key difference seems to be inc losing the old image or not. Perhaps we have a function that takes two arguments latest_image_ and previous_image and sometimes previous_image is null and we don't call close?

Reply:
I refactored this into a method that only handles the hardware buffer. Closing the hardware buffer and potentially the image is left to the caller, which I think is cleaner and reduces the duplication.

Reply:
Thanks!